cpp/include/cudf/functions.h

```diff
@@ -894,6 +894,60 @@ gdf_error gdf_group_by_count(int ncols,                    // # columns
                              gdf_column* out_col_agg,      //aggregation result
                              gdf_context* ctxt);            //struct with additional info: bool is_sorted, flag_sort_or_hash, bool flag_count_distinct
 
+/* --------------------------------------------------------------------------*/
+  /**
+   * @brief Given a set of columns, of which a subset of these are defined to be the group by columns,
+   * all columns are sorted by the group by columns and returned, along with a column containing
+   * a list of the start indices of each group
+   *
+   * @Param[in] The number of columns in the dataset
+   * @Param[in] The input columns in the dataset
+   * @Param[in] The number of columns to be grouping by
+   * @Param[in] The column indices of the input dataset that will be grouped by
+   * @Param[out] The dataset sorted by the group by columns (needs to be pre-allocated)
+   * @Param[out] A column containing the starting indices of each group. Indices based off of new sort order. (needs to be pre-allocated)
+   * @Param[in] The context used to control how nulls are treated in a sort and in group by
+   *   context->flag_nulls_sort_behavior< 0 = Nulls are are treated as largest, 
+   *   1 = Nulls are treated as smallest, 2 = Special multi-sort case any row with null is largest>
+   *   context-> flag_groupby_include_nulls <0 = Nulls are ignored in group by keys (Pandas style), 
+                        1 = Nulls are treated as values in group by keys where NULL == NULL (SQL style)>
+   *
+   * @Returns gdf_error with error code on failure, otherwise GDF_SUCESS
+   */
+  /* ----------------------------------------------------------------------------*/
+gdf_error gdf_group_by_wo_aggregations(int num_data_cols,
```

Comment:
So, I've looked at this function's Doxygen comment, and - this does not perform GROUP BY. It is merely an out-of-place sort.

Also:
1. Shorthands are confusing for many people who can't guess what you meant. Use "without" rather than "wo".
2. The type of `num_data_cols` should not be a plain `int`, we should have a type definition for a column index / number-of-columns.